### PR TITLE
nep1: guidelines for getters and setters

### PR DESCRIPTION
**Pull request by timotheecour**
refs https://github.com/nim-lang/Nim/pull/17573#issuecomment-810633196

### CODE REVIEW

File: doc/nep1.rst [hunk 1]
```diff
@@ -301,3 +301,13 @@ Miscellaneous
     let a = """foo
     bar
     """
+
+- A getter API for a private field `foo` should preferably be named `foo`, not `getFoo`.
+  A getter-like API should preferably be named `getFoo`, not `foo` if:
+    * the API has side effects
+    * or the cost is not `O(1)`
+  For in between cases, there is no clear guideline.
```
Comment (juancarlospaco) [inline doc/nep1.rst:309]: Should be better to think on a clear guideline, because this is the official documentation... :thinking:

Comment (ghost) [reply doc/nep1.rst:309]: @juancarlospaco you can't decide for all cases, Nim is not the "one way to do it" language

Comment (juancarlospaco) [reply doc/nep1.rst:309]: I feel the `O(1)` may be too specific, maybe wording it like `if it is computationally costly name it getFoo` or similar.

File: doc/nep1.rst [hunk 2]
```diff
@@ -301,3 +301,13 @@ Miscellaneous
...
+  In all cases (`foo` or `getFoo`), the 1st argument should be called `self`.
+
+- Likewise with a setter API, replacing `foo` with `=foo` and `getFoo` with `setFoo`
```
Comment (metagn) [inline doc/nep1.rst]:
```suggestion
- Likewise with a setter API, replacing `foo` with `foo=` and `getFoo` with `setFoo`
```

File: doc/nep1.rst [hunk 3]
```diff
@@ -301,3 +301,13 @@ Miscellaneous
     let a = """foo
     bar
     """
+
+- A getter API for a private field `foo` should preferably be named `foo`, not `getFoo`.
+  A getter-like API should preferably be named `getFoo`, not `foo` if:
+    * the API has side effects
+    * or the cost is not `O(1)`
+  For in between cases, there is no clear guideline.
+  In all cases (`foo` or `getFoo`), the 1st argument should be called `self`.
```
Comment (Araq) [inline doc/nep1.rst]: No. Hardly any code does it via `self` and it's not important.

Comment (timotheecour) [reply doc/nep1.rst]: i've removed this line from this PR to unblock it

File: doc/nep1.rst (hunk 2 above)
Comment (timotheecour) [reply doc/nep1.rst]: good point, thanks! fixed

File: doc/nep1.rst (hunk 1 above)
Comment (timotheecour) [reply doc/nep1.rst:309]: That's the whole point, O(1) vs not O(1) though. Builtin field accessors are O(1), and so should procs mascarading as field accessors.

**Pull request merged**